table/block_based_table_reader.cc

```diff
@@ -1618,6 +1618,19 @@ BlockBasedTable::BlockEntryIteratorState::NewSecondaryIterator(
           &rep->internal_comparator, nullptr, true, rep->ioptions.statistics);
     }
   }
+
+  if (num_file_reads_ < 2) {
+    num_file_reads_++;
+  } else if (handle.offset() + static_cast<size_t>(handle.size()) + kBlockTrailerSize > readahead_limit) {
+    num_file_reads_++;
+    // Do not readahead more than 16MB.
+    readahead_size = std::min(size_t{16 * 1024 * 1024}, readahead_size);
+    table_->rep_->file->Prefetch(handle.offset(), readahead_size);
```

Comment:
if Prefetch is called a bit before running out of data (so reader never has to wait), do we need such a large readahead size?

Reply:
Good point. Maybe 1MB or so is more than enough.

Reply:
Maybe 512KB.

Reply:
I did more experiments and collected data to get a sense for which readahead size gives us the best performance: https://gist.github.com/sagar0/fcba5c2c2d6592929078a295da583d11. Look at the Avg MB/s column (average of 3 runs). Seems like 256KB is the one to go with.

Reply:
Note that these experiments were run on an SSD. But for HDDs, I am assuming a larger readahead size gives a better performance.